engine/src/main/java/org/hibernate/search/store/impl/FSMasterDirectoryProvider.java

```diff
@@ -223,11 +205,9 @@ public void run() {
 					log.unableToSynchronizeSource( indexName, e );
 					return;
 				}
-				if ( ! new File( destination, CURRENT_DIR_NAME[oldIndex] ).delete() ) {
-					log.unableToRemovePreviousMarket( indexName );
-				}
 				try {
-					new File( destination, CURRENT_DIR_NAME[index] ).createNewFile();
+					Files.delete( destination.resolve( CURRENT_DIR_NAME[oldIndex] ) );
```

Comment:
that's right. The File API doesn't throw an error when it fails, it would just return false.
The NIO API throws appropriate exceptions.